rendercv/cli/commands.py

```diff
@@ -164,17 +184,8 @@ def cli_command_render(
     # 4. render PNG files from the PDF
     # 5. generate the Markdown file
     # 6. render the Markdown file to a HTML (for Grammarly)
-    number_of_steps = 6
-    if dont_generate_png:
-        number_of_steps = number_of_steps - 1
-    if dont_generate_markdown:
-        # if the Markdown file is not generated, then the HTML file is not generated
-        number_of_steps = number_of_steps - 2
-    else:
-        if dont_generate_html:
-            number_of_steps = number_of_steps - 1
-
-    with printer.LiveProgressReporter(number_of_steps) as progress:
+    initial_steps = 1
```

Comment:
Apparently, updating render settings after the YAML file is read complicates some logic. I would try to avoid complicating the logic as much as possible. Here is my proposal:

Before starting the steps, do these first:

-   Figure out which settings are given from CLI, i.e., which are different from the default values.
-   Read the YAML file as a dict, and open the settings as a dict. We don't need to validate yet.
-   Add/overwrite all the given CLI settings to the settings dict, which is read from the YAML file.
-   Then, compute the steps using the old logic and start the steps (start directly from validation).

Reply:
Moved the reading logic to the top.

Now our 1st step only verifies the dictionary. Not reads them.